tck/src/main/java/org/eclipse/microprofile/rest/client/tck/InvokeWithJsonPProviderTest.java

```diff
@@ -102,7 +102,9 @@ private void testGet(JsonPClient client, String clientType) {
         reset();
         stubFor(get(urlEqualTo("/"))
             .willReturn(aResponse()
-                .withBody("[{\"key\": \"value\"}, {\"key\": \"anotherValue\"}]")));
+                .withHeader("Content-Type", "application/json")
```

Comment:
`Content-Type` should be immaterial to this test.

Reply:
I don't know if this anchored in the HTTP or JAX-RS specs, but without a valid `Content-type` header my underlying JAX-RS implementation does not pick up the JSON  reader.

Reply:
Ah ha.  You should be activating the JSON Reader simply because the method returns a `JsonArray`

Reply:
BTW, the spec mentions "OR" between the produces/consumes/method return values.  [Take a look.](https://github.com/eclipse/microprofile-rest-client/blob/master/spec/src/main/asciidoc/providers.asciidoc#json-p-provider)

Reply:
I don't want lengthy arguments about this, but generally speaking content-type should be present in HTTP. This doesn't change the nature of the test, but rather refines it, so why not simply include it? 

The impl we leverage (RESTEasy) chokes on it, and this would be a simple fix, which I think is fully aligned with the test assertion and the HTTP spec.

Reply:
@heiko-braun i understand your point.  I would recommend that we change the `Content Type` to be `text/plain` to ensure that it's being activated based on the object in use, not the media of the server response.

Reply:
I believe I get the idea, however in RESTEasy this leads to `Unable to find a MessageBodyReader of content-type text/plain and type interface javax.json.JsonObject`

Reply:
FWIW, I'm fine with the final resolution of this however I have an application running today that uses resteasy, receives content type of text/html and deserializes it as JSON.  It's all about fixing the produces/consumes on the MBR/MBW.

Reply:
Ok, so I ended up digging into the spec a bit to understand what in the world resteasy is doing.

I found this commit: https://github.com/resteasy/Resteasy/commit/fb33d3c1bac91bdb1d686008bb679fae7044820d

It references sections of the spec.  So I opened up the spec, and found these notes:

> • If the method is annotated with @Produces, set P = {V (method)} where V (t) represents the
values of @Produces on the specified target t.
> • Else if the class is annotated with @Produces, set P = {V (class)}.
> • Else set P = {V (writers)} where ‘writers’ is the set of MessageBodyWriter that support the
class of the returned entity object.

Since the interface doesn't define any [https://github.com/eclipse/microprofile-rest-client/blob/master/tck/src/main/java/org/eclipse/microprofile/rest/client/tck/interfaces/JsonPClient.java](produces/consumes), it would appear that resteasy is doing the wrong thing and looking at the content type to determine the match.  Based on the spec content I copied above, I would expect that it falls into bullet 3 and uses the class of the returned entity to be invoked (noting that in fact we should be using MBR instead of MBW in the type safe client).

If you like, I can create a ticket against resteasy.